src/controller/codingPlatform/codingPlatform.service.ts

```diff
@@ -332,6 +334,191 @@ export class CodingPlatformService {
       return {...results[0], questionInfo: data[0], shapecode}
     } catch (error){
       throw error
-    }
-  }
+    }
+  }
+
+  async getLanguagesById(languageId: string) {
+    try {
+      const language = await db.select().from(zuvyLanguages).where(sql`${zuvyLanguages.id} = ${languageId}`);
+      return language;
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  async createCodingQuestion(createCodingQuestionDto: any) {
+    const { testCases, ...questionData } = createCodingQuestionDto;
+    try {
+      const question:any = await db.insert(zuvyCodingQuestions).values(questionData).returning();
+      let testCaseAndExpectedOutput = [];
+      for (let i = 0; i < testCases.length; i++) {
+        testCaseAndExpectedOutput.push({ questionId: question[0].id, ...testCases[i] });
+      }
+      await db.insert(zuvyTestCases).values(testCaseAndExpectedOutput);
+      return question;
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  async updateCodingQuestion(id: number, updateCodingQuestionDto: any) {
+    const { testCases, expectedOutputs, ...questionData } = updateCodingQuestionDto;
+    try {
+      const question = await db.update(zuvyCodingQuestions).set(questionData).where(sql`${zuvyCodingQuestions.id} = ${id}`).returning();
+      if (testCases && expectedOutputs) {
+        await db.update(zuvyTestCases).set({ testCases, expectedOutputs }).where(sql`${zuvyTestCases.questionId} = ${id}`);
+      }
+      return question;
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  async deleteCodingQuestion(id: number) {
+    try {
+      await db.delete(zuvyCodingQuestions).where(sql`${zuvyCodingQuestions.id} = ${id}`);
+      return { message: 'Coding question deleted successfully' };
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  async createLanguage(createLanguageDto: any) {
+    try {
+      const language = await db.insert(zuvyLanguages).values(createLanguageDto);
+      return language;
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  async getCodingQuestion(id: number) {
+    try {
+      const question = await db.select().from(zuvyCodingQuestions).where(sql`${zuvyCodingQuestions.id} = ${id}`);
+      const testCases = await db.select().from(zuvyTestCases).where(sql`${zuvyTestCases.questionId} = ${id}`);
+      let templates = await this.generateTemplates(question[0].title, testCases[0]);
+      return { question, testCases, templates };
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  async generateTemplates(functionName, parameters) {
```

Comment:
Here also,with increase in language,the hardcoded part will also be even lengthy.

Reply:
Yep.